src/module.rs

```diff
@@ -61,14 +56,30 @@ pub fn init(_module: ModuleContext) -> bool {
         service_instance, skywalking_version, "Starting skywalking agent"
     );
 
+    let worker_addr = {
+        match tempfile::NamedTempFile::new() {
```

Comment:
The `NamedTempFile` will be deleted after the `init` function finished.

How about wrap it with `ManuallyDrop` and manual drop it in `shutdown` function?

Reply:
We do not actually need a file here, but a name. I've not found a proper `tempnam`, and `NamedTempFile` happens fit. `tempnam` has its own problem with racing, and a dropped `NamedTempFile` has no luck either.
If the file does exist, `bind` would fail with `Address already in use (os error 48)`.

Reply:
I think the socket file is need, Because `UnixStream::connect` is happen every `report` which connect not occurred. For php-fpm using `dynamic` or `ondemand`, if the socket file not exists, the new worker process do `UnixStream::connect` will failed.

Reply:
You are right about `UnixStream::connect`, and `UnixListener::bind` will create the socket file for us. `NamedTempFile` creates a regular file that cannot be connected to either. Remember we've already passed the tests.